modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java

```diff
@@ -56,23 +56,22 @@ public String getOAuth2AccessToken(String username, String password, String clie
         headers.add(HttpHeaders.CONTENT_TYPE, TOKEN_CONTENT_TYPE);
 
         HttpEntity<String> httpEntity = new HttpEntity<>(body, headers);
-        ResponseEntity<String> response;
         try {
-            response = doRestCall(getEndpoint(), HttpMethod.POST, httpEntity, String.class);
+            ResponseEntity<String> response = doRestCall(getEndpoint(), HttpMethod.POST, httpEntity, String.class);
+
+            if (response.getStatusCode() == HttpStatus.OK) {
```

Comment:
HEre I propose a method checkStatus(condition) throwing an IllegalStateException which is caught in the big catch below again.

Apply this pattern to the other instances of this sequence as well.

Or even better, do not do an else branch but return the exception at the end of the method or the itemToModify